torch/onnx/symbolic_helper.py

```diff
@@ -449,6 +449,8 @@ def _topk_helper(g, input, k, dim, largest=True, sorted=False, out=None):
         k = g.op("Constant", value_t=torch.tensor([k], dtype=torch.int64))
     else:
         k = _reshape_helper(g, k, g.op("Constant", value_t=torch.tensor([1])))
+        if _try_get_scalar_type(k) == "Int":
```

Comment:
What about
`if k.type().scalarType() != "Long":`?

Two differences from what you have:
* Allows casting other types, e.g. `"Char"`. Any reason not to do that?
* Don't use `_try`, just access the scalar type directly. Unless there is a situation where we expect `k` to not have a scalar type here?

Reply:
Agree with the first point. For second point I feel the graph never guarantees presence of shape nor scalar type, not using `_try` gives a slight chance of runtime error. Maybe we should add graph enforcement of every value having scalar type?
I will go with `if _try_get_scalar_type(k) != "Long":` as it produces correct graph even if scalar type is missing.

Reply:
I don't understand how the scalar type could be missing but we still end up with a valid graph in the end. Could you explain how that might happen?
In any case, this is fine, feel free to merge as-is but if you wouldn't mind educating me I'd appreciate it.